test/pqnnlp.jl

```diff
@@ -80,3 +80,23 @@ end
   @test ges_psenlp.status == :first_order
   @test ges_psnlp.status == :first_order
 end
+
+
+@testset "Merge, from a limit function" begin
```

Comment:
The PR looks good, I think it misses tests when `merging=false` ? or I miss something.

Reply:
Your correct, but the code related to `merging=false` is tested, since most of partially-separable functions do not have a partitioned structure triggering the merge step.
I made this function to prevent the merge, in case the user does not want it explicitely, but with the soft condition I put I do note recommend it ;)